packages/insomnia-app/app/ui/redux/__tests__/selectors.test.ts

```
@@ -28,19 +29,19 @@ describe('selectors', () => {
       const state = await reduxStateForTest({ activeSpaceId: 'some-other-space' });
 
       const space = selectActiveSpace(state);
-      expect(space).toBeUndefined();
+      expect(space).toStrictEqual(expect.objectContaining<Partial<Space>>({ _id: BASE_SPACE_ID }));
     });
 
     it('should return undefined if no active space', async () => {
       // create two spaces
       await models.space.create();
       await models.space.create();
 
-      // set first as selected
-      const state = await reduxStateForTest({ activeSpaceId: null });
+      // set base as selected
+      const state = await reduxStateForTest({ activeSpaceId: undefined });
 
       const space = selectActiveSpace(state);
-      expect(space).toBeUndefined();
+      expect(space).toStrictEqual(expect.objectContaining<Partial<Space>>({ _id: BASE_SPACE_ID }));
```

Tests to make sure the base space is returned in cases where we can't find the actual expected space

---

with the context of where we were, where we are, and where we're going, this makes sense.  on the other hand if I step back and think about this it seems like this should throw an error in some way.  what is the (valid) situation in which there's no active space?

---

The only way someone can reach this situation is if they manually fiddle with the space value in local storage, there's no valid way someone can reach this. And we could show an error but working case by case, in this particular case we are able to self-recover (by showing the base space), so IMO we should self-recover.